src/semantics/tests/infer.rs

```diff
@@ -481,3 +481,30 @@ fn proj_weird() {
         parse_normalize(&mut codemap, &context, expected_ty),
     );
 }
+
+#[test]
+fn array() {
+    let mut codemap = CodeMap::new();
+    let context = Context::default();
+
+    let expected_ty = r"Array 2 I32";
+    let given_expr = r#"[1 : I32; 2]"#;
+
+    assert_term_eq!(
+        parse_infer(&mut codemap, &context, given_expr).1,
+        parse_normalize(&mut codemap, &context, expected_ty),
+    );
+}
+
+#[test]
+fn array_ambiguous() {
+    let mut codemap = CodeMap::new();
+    let context = Context::default();
+
+    let given_expr = r#"[1 : I32; 2]"#;
+
+    match infer(&context, &parse(&mut codemap, given_expr)) {
+        Err(_) => unimplemented!("error"),
```

Comment:
Both arms of the match panic. Should this one just return?

Reply:
Still a WIP - I'll be adding proper errors here!

Reply:
I kind of find errors a bit hard/annoying to make, so I leave them till last 🤦‍♂️ - I think its mainly because I have to think up a new name for the error type...